manifests/server/rsync_shares.pp

```diff
@@ -49,78 +53,101 @@
 
-      rsync::server::section { "openldap_server_${_env}":
-        auth_users  => ["openldap_rsync_${_env}"],
-        comment     => "Configuration for OpenLDAP for Environment: ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/openldap/server",
-        hosts_allow => $_trusted_nets
-      }
+      # Do the Global items first
+      if $rsync_environments[$_env]['rsync']['global'] {
+        $_globals_dir = 'rsync/Global'
 
-      rsync::server::section { "bind_dns_default_${_env}":
-        auth_users  => ["bind_dns_default_rsync_${_env}"],
-        comment     => "Default DNS configurations for named for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/bind_dns/default",
-        hosts_allow => $_trusted_nets
-      }
+        if 'clamav' in $rsync_environments[$_env]['rsync']['global']['shares'] {
+          rsync::server::section { "clamav_${_env}":
+            comment     => "ClamAV Virus Database Updates for Environment ${_env}",
+            path        => "${rsync_base}/${_env}/${_globals_dir}/clamav",
+            hosts_allow => $_trusted_nets
+          }
+        }
 
-      rsync::server::section { "apache_${_env}":
-        auth_users     => ["apache_rsync_${_env}"],
-        comment        => "Apache configurations for Environment ${_env}",
-        path           => "${rsync_base}/${_env}/${_rsync_subdir}/apache",
-        hosts_allow    => ['127.0.0.1'],
-        outgoing_chmod => 'o-rwx'
-      }
+        if 'mcafee' in $rsync_environments[$_env]['rsync']['global']['shares'] {
+          rsync::server::section { "mcafee_${_env}":
+            comment     => "McAfee DAT files for Environment ${_env}",
+            path        => "${rsync_base}/${_env}/${_globals_dir}/mcafee",
+            hosts_allow => $_trusted_nets
+          }
+        }
 
-      rsync::server::section { "tftpboot_${_env}":
-        auth_users  => ["tftpboot_rsync_${_env}"],
-        comment     => "Tftpboot server configurations for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/tftpboot",
-        hosts_allow => $_trusted_nets
+        if 'jenkins_plugins' in $rsync_environments[$_env]['rsync']['global']['shares'] {
+          rsync::server::section { "jenkins_plugins_${_env}":
+            comment     => "Jenkins Configuration for Environment ${_env}",
+            path        => "${rsync_base}/${_env}/${_globals_dir}/jenkins_plugins",
+            hosts_allow => $_trusted_nets
+          }
+        }
       }
 
-      rsync::server::section { "mcafee_${_env}":
-        comment     => "McAfee DAT files for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/mcafee",
-        hosts_allow => $_trusted_nets
-      }
+      # OS Specific Items
+      (keys($rsync_environments[$_env]['rsync']) - ['global','id']).each |String $_os| {
+        $_os_id = $rsync_environments[$_env]['rsync'][$_os]['id']
```

Comment:
It doesn't like this because there is no 'id' value for simp_rsync_environments under the OS level.  The 'id' is under the major version level of the hash.  I wasn't sure if you wanted to add an id or just use the os name for _os_id.  (I used the os name and it got to the next error which is in the pam module.)